### PR TITLE
[MNT] Move linting to ruff

### LINKED ISSUE #1689: [MNT] move linting to `ruff`, resolve strange linting behaviour
We should move the linting to `ruff` - good first issue for a maintenance affine contributor.

There are also some strange behaviours in the precommits which I cannot pinpoint, e.g., the linter insisting on long single lines instead of line breaks where they would be appropriate. Though I would hope this is resolved as a side effect of moving to `ruff`.

### PR DESCRIPTION
**Pull request by airookie17**
### Description

This PR replaces the existing linters `flake8` and `isort` with `ruff`. 

Fixes #1689

### Checklist

- [x] Linked issues (if existing)
- [ ] Amended changelog for large changes (and added myself there as contributor)
- [ ] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.
      To run hooks independent of commit, execute `pre-commit run --all-files`

Make sure to have fun coding!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,34 @@
+[tool.ruff]
+line-length = 120
```
Comment (fkiraly) [inline pyproject.toml:2]: could you change this to 88?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks good! I only have a comment about the max line length.

**Review by fkiraly — APPROVED** [review]
Actually, I am happy with this since it is an 1:1 translation.

We should change line length in another PR.

**Comment by airookie17** [comment]
Thanks for the review! I will change the line length in a different PR as suggested.

**Pull request merged**